tests/unit/grains/test_core.py

```diff
@@ -1227,33 +1258,27 @@ def _check_type(key, value, ip4_empty, ip6_empty):
                 self._check_empty(key, value, ip6_empty)
                 self._check_ipaddress(value, ip_v="6")
 
-        ip4_mock = [(2, 1, 6, "", (IP4_ADD1, 0)), (2, 3, 0, "", (IP4_ADD2, 0))]
-        ip6_mock = [
-            (10, 1, 6, "", (IP6_ADD1, 0, 0, 0)),
-            (10, 3, 0, "", (IP6_ADD2, 0, 0, 0)),
-        ]
-        if ip6_empty:
-            getaddrinfo_side_effect = [ip4_mock, ip4_mock, ip6_mock]
-        elif ip4_empty:
-            getaddrinfo_side_effect = [ip4_mock, ip6_mock, ip6_mock]
-        else:
-            getaddrinfo_side_effect = [ip4_mock, ip6_mock]
-
         with patch.object(
             salt.utils.network, "ip_addrs", MagicMock(return_value=net_ip4_mock)
         ), patch.object(
             salt.utils.network, "ip_addrs6", MagicMock(return_value=net_ip6_mock),
         ), patch.object(
-            core.socket,
-            "getaddrinfo",
-            side_effect=getaddrinfo_side_effect,
-            autospec=True,
+            core.socket, "getaddrinfo", side_effect=_getaddrinfo,
         ):
             get_fqdn = core.ip_fqdn()
             ret_keys = ["fqdn_ip4", "fqdn_ip6", "ipv4", "ipv6"]
             for key in ret_keys:
                 value = get_fqdn[key]
                 _check_type(key, value, ip4_empty, ip6_empty)
+                if key.startswith("fqdn_ip"):
+                    if key.endswith("4"):
+                        self._check_ip_fqdn_set(
+                            value, ip4_empty, _set=[IP4_ADD1, IP4_ADD2]
+                        )
+                    if key.endswith("6"):
+                        self._check_ip_fqdn_set(
+                            value, ip6_empty, _set=[IP6_ADD1, IP6_ADD2]
+                        )
```

Comment:
**question**: what are the valid possibilities here? 4 or 6? 4 *and* 6? neither?

Actually, now that I'm looking a bit closer, it may be that this should just be:

```
value = get_fqdn["fqdn_ipv4"]
_check_type("fqdn_ipv4", value, ipv4_empty, ipv6_empty)
self._check_ip_fqdn_set(value, ip4_empty, _set=[IP4_ADD1, IP4_add2])
``` 
and then a similar statement for ipv6? Does that make sense?